Pick & Place Exercise (Gazebo + Console + Rviz)  with web-templates

predator4hack opened the pull request.

predator4hack committed:
commented the rsync command to create a docker image

predator4hack committed:
Docker Mount is working

predator4hack committed:
Created an entry for pic and place in db.sqlite3

predator4hack committed:
Added exercise.py and assets

predator4hack committed:
exercise.html

predator4hack committed:
Added teaser png

predator4hack committed:
Added ace-builds

predator4hack committed:
Added Code, exercise and launch files

predator4hack committed:
added kinematic widget in html page

predator4hack committed:
Updated HAL and created env functions

predator4hack committed:
Added GAZEBO_RESOURCE_PATH in manager

predator4hack committed:
Updated launcher.js

predator4hack committed:
configured repo for IndustrialRobots

predator4hack committed:
updated dockerfile

predator4hack committed:
replaced my repo in dockerfile

predator4hack committed:
removed rsync

predator4hack committed:
Merge branch 'JdeRobot:master' into pick_place

predator4hack committed:
Made changes for new Image

predator4hack committed:
Updated ROS_PACKAGE_PATH in dockerfile

predator4hack committed:
updated dockerfile

predator4hack committed:
Updated world path

predator4hack committed:
added rospkg library

predator4hack committed:
Added Comments

predator4hack committed:
Added widget

predator4hack committed:
Connection Resolved

predator4hack committed:
Connection Working

predator4hack committed:
Added child css

predator4hack committed:
connections are working and hal and env APIs are created

predator4hack committed:
Css Working Now

predator4hack committed:
Fixed merge conflicts

predator4hack committed:
Added rviz display

predator4hack committed:
fixed some bug

predator4hack committed:
Rviz is working nowpwd

predator4hack committed:
Merge branch 'Rviztesting' into pick_place

Rviz is incorporated in the exercise

predator4hack committed:
HAL apis

predator4hack committed:
Rviz objects spawned properly

predator4hack committed:
Added kinematic perspective

predator4hack committed:
Added start.sh for the config of workspace

predator4hack committed:
Fixed simulation bugs

predator4hack committed:
Updated the dockerfile

predator4hack committed:
Fixed issues with reset button

predator4hack committed:
Fixed the templates issue with other exercises

predator4hack committed:
Removed redundant exercises

predator4hack committed:
Added plugin path for acceleration enabled containers

predator4hack commented:
solves issue [#1133](https://github.com/JdeRobot/RoboticsAcademy/issues/1133)

sixer51 commented on scripts/Dockerfile:
```diff
@@ -2,17 +2,19 @@ FROM jderobot/base
 # RoboticsAcademy
-RUN git clone https://github.com/JdeRobot/RoboticsAcademy.git -b master
-RUN rsync -a --exclude 'ace-builds' /RoboticsAcademy/exercises/static/exercises/* /RoboticsAcademy/exercises
+RUN git clone https://github.com/predator4hack/RoboticsAcademy.git -b pick_place
```
This line should not be changed

sixer51 commented on scripts/Dockerfile-2.4:
```diff
@@ -6,27 +6,31 @@ RUN mkdir -p /opt/jderobot && cd /opt/jderobot && \
 
 # RoboticsAcademy
-RUN git clone https://github.com/JdeRobot/RoboticsAcademy.git -b master
+RUN git clone https://github.com/predator4hack/RoboticsAcademy.git -b pick_place
```
his line should not be changed

sixer51 commented on scripts/Dockerfile:
```diff
@@ -2,17 +2,19 @@ FROM jderobot/base
-RUN git clone https://github.com/JdeRobot/RoboticsAcademy.git -b master
-RUN rsync -a --exclude 'ace-builds' /RoboticsAcademy/exercises/static/exercises/* /RoboticsAcademy/exercises
+RUN git clone https://github.com/predator4hack/RoboticsAcademy.git -b pick_place
+# RUN rsync -a --exclude 'ace-builds' /RoboticsAcademy/exercises/static/exercises/* /RoboticsAcademy/exercises
```
Why is this line commented in Dockerfile, but kept in Dockerfile-2.4?

sixer51 committed:
Delete settings.json for vscode

sixer51 commented on scripts/Dockerfile:
```diff
@@ -2,17 +2,19 @@ FROM jderobot/base
 RUN mkdir -p /opt/jderobot && cd /opt/jderobot && \ 
-	git clone -b melodic-devel https://github.com/JdeRobot/CustomRobots.git
+	git clone -b melodic-devel https://github.com/JdeRobot/CustomRobots.git && \
+	git clone https://github.com/JdeRobot/IndustrialRobots.git
```
It seems that adding `-b web-template` is necessary. If the industrial robot exercise is still not working properly for RADI2.3. All modifications in this file and other RADI2.3 related files should be recovered.

predator4hack reviewed (commented):
Yeah,, ig any changes inside the dockerfile is irrelevant because there is no catkin workspace in it. I'll remove all the changes inside dockerfile.q11

pushkalkatara commented on exercises/pick_place/web-template/gui.py:
```diff
@@ -0,0 +1,247 @@
+        # Acknowledge Message for GUI Thread
+        if message[:4] == "#ack":
+            self.set_acknowledge(True)
+        elif message[:4] == "#tur":
```
If we are referring turtlebot template to make the pick and place exercise work, please make sure to update variable names / code specific to pick_place. 

There is a lot of turtle bot exercise specific redundant code which came for pick_place exercise as well. Please find those specific pieces and update the exercise.

pushkalkatara commented on exercises/static/exercises/pick_place/web-template/hal.py:
```diff
@@ -0,0 +1,84 @@
+        return new_instance
+    
+    # Get Image from ROS Driver Camera
+    def get_frontal_image(self):
```
Do we need all these functions for pick_place exercise?

pushkalkatara commented on scripts/manager.py:
```diff
@@ -8,7 +8,7 @@
 import json
 import stat
-
+import manager_utils as m_utils
```
All specific changes to scripts/manager.py would be a feature accessible to all exercises in JdeRobot Academy.
In this case, it is the addition of the Rviz plugin. I guess other projects would also be making similar changes. 

Maybe this should go as a separate feature-issue-pr into JdeRobotAcademy and not part of the pick_place exercise addition PR.

@sixer51 @jmplaza  what are your thoughts on the same?

predator4hack replied:
Actually, Siddharth has already created a pull request related to this: https://github.com/trunc8/RoboticsAcademy/commits/issue-801

pushkalkatara commented:
I reproduced the branch on my system and the exercise works perfectly. Thanks Chandan (@predator4hack) for your contribution.

The number of changes in the current PR (1931) is daunting to review. Probably most of the changes are due to the static files of ace-builds for the website. Keeping those files for new exercise is a necessity I guess. 

Maybe we can break the PR and get the raw template merged in JdeRobot:master first, in another PR, we can update changes specific to `pick_place` exercise. This way no. of changes in files would be less and code-review would be easier.

pushkalkatara commented:
Also, the current docker build is from scratch, It took around 2 hours on my system for the build to complete. 

Can we somehow use RoboticsAcademy current images as our Dockerfile's base image, and have a Dockerfile specifically to build the exercise? this way build time would be shorter I guess.

predator4hack commented:
> I reproduced the branch on my system and the exercise works perfectly. Thanks Chandan (@predator4hack) for your contribution.
> 
> The number of changes in the current PR (1931) is daunting to review. Probably most of the changes are due to the static files of ace-builds for the website. Keeping those files for new exercise is a necessity I guess.
> 
> Maybe we can break the PR and get the raw template merged in JdeRobot:master first, in another PR, we can update changes specific to `pick_place` exercise. This way no. of changes in files would be less and code-review would be easier.

Yeah, I guess we can do that

predator4hack commented:
> Also, the current docker build is from scratch, It took around 2 hours on my system for the build to complete.
> 
> Can we somehow use RoboticsAcademy current images as our Dockerfile's base image, and have a Dockerfile specifically to build the exercise? this way build time would be shorter I guess.

Yeah, we have multi-stage build for this. Once you create the base image for any version, you can build again on top of it. And you don't need to build the base image again and again, which takes most of the time.

predator4hack committed:
Removed the changes made for RADI2.3

Industrial Robot exercise doesn't work with RADI2.3 because of the unavailability of the catkin_ws. So the changes for the Dockerfile(RADI2.3) are unnecessary.

predator4hack committed:
Updated the repo from personal fork to JdeRobot
